StudentProfilePageUiTest fails on live server

org.openqa.selenium.ElementNotVisibleException: Element is not currently visible and so may not be interacted with
Command duration or timeout: 6 milliseconds
Build info: version: '2.46.0', revision: '87c69e2', time: '2015-06-04 16:16:47'
System info: host: 'damithch-d9010', ip: '172.26.184.184', os.name: 'Windows 7', os.arch: 'amd64', os.version: '6.1', java.version: '1.7.0_45'
Session ID: 024bb8b7-e38a-4060-ae5e-794214fc1929
Driver info: org.openqa.selenium.firefox.FirefoxDriver
Capabilities [{platform=WINDOWS, acceptSslCerts=true, javascriptEnabled=true, cssSelectorsEnabled=true, databaseEnabled=true, browserName=firefox, handlesAlerts=true, nativeEvents=false, webStorageEnabled=true, rotatable=false, locationContextEnabled=true, applicationCacheEnabled=true, takesScreenshot=true, version=41.0.2}]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:204)
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:156)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:605)
	at org.openqa.selenium.remote.RemoteWebElement.execute(RemoteWebElement.java:269)
	at org.openqa.selenium.remote.RemoteWebElement.click(RemoteWebElement.java:80)
	at teammates.test.pageobjects.AppPage.click(AppPage.java:430)
	at teammates.test.cases.ui.browsertests.StudentProfilePageUiTest.testContent(StudentProfilePageUiTest.java:81)
	at teammates.test.cases.ui.browsertests.StudentProfilePageUiTest.allTests(StudentProfilePageUiTest.java:42)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:714)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:334)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:329)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291)
	at org.testng.SuiteRunner.run(SuiteRunner.java:240)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:111)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:204)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:175)
Caused by: org.openqa.selenium.ElementNotVisibleException: Element is not currently visible and so may not be interacted with
Build info: version: '2.46.0', revision: '87c69e2', time: '2015-06-04 16:16:47'
System info: host: 'damithch-d9010', ip: '172.26.184.184', os.name: 'Windows 7', os.arch: 'amd64', os.version: '6.1', java.version: '1.7.0_45'
Driver info: driver.version: unknown
	at <anonymous class>.fxdriver.preconditions.visible(file:///C:/Users/dcsdcr/AppData/Local/Temp/anonymous2765636981848031879webdriver-profile/extensions/fxdriver@googlecode.com/components/command-processor.js:9615)
	at <anonymous class>.DelayedCommand.prototype.checkPreconditions_(file:///C:/Users/dcsdcr/AppData/Local/Temp/anonymous2765636981848031879webdriver-profile/extensions/fxdriver@googlecode.com/components/command-processor.js:12265)
	at <anonymous class>.DelayedCommand.prototype.executeInternal_/h(file:///C:/Users/dcsdcr/AppData/Local/Temp/anonymous2765636981848031879webdriver-profile/extensions/fxdriver@googlecode.com/components/command-processor.js:12282)
	at <anonymous class>.DelayedCommand.prototype.executeInternal_(file:///C:/Users/dcsdcr/AppData/Local/Temp/anonymous2765636981848031879webdriver-profile/extensions/fxdriver@googlecode.com/components/command-processor.js:12287)
	at <anonymous class>.DelayedCommand.prototype.execute/<(file:///C:/Users/dcsdcr/AppData/Local/Temp/anonymous2765636981848031879webdriver-profile/extensions/fxdriver@googlecode.com/components/command-processor.js:12229)

--- a/src/test/java/teammates/test/cases/ui/browsertests/StudentProfilePageUiTest.java
+++ b/src/test/java/teammates/test/cases/ui/browsertests/StudentProfilePageUiTest.java
@@ -108,6 +108,7 @@ public class StudentProfilePageUiTest extends BaseUiTestCase {
 
         // Verify with retry after upload picture due to inconsistency of .click in detecting page load
         profilePage.verifyStatusWithRetry(Const.StatusMessages.STUDENT_PROFILE_PICTURE_SAVED, 10);
+        profilePage.waitForUploadEditModalVisible();
         profilePage.verifyHtmlMainContent("/studentProfilePageFilled.html");
         
         profilePage.click(By.className("close"));

--- a/src/test/java/teammates/test/pageobjects/StudentProfilePage.java
+++ b/src/test/java/teammates/test/pageobjects/StudentProfilePage.java
@@ -103,7 +103,7 @@ public class StudentProfilePage extends AppPage {
 
     public void showPictureEditor() throws Exception {
         uploadPopupButton.click();
-        waitForElementVisibility(browser.driver.findElement(By.id("studentPhotoUploader")));
+        waitForUploadEditModalVisible();
     }
 
     public void fillShortName(String shortName) {
@@ -214,6 +214,10 @@ public class StudentProfilePage extends AppPage {
     public void verifyUploadButtonState(boolean expectedState) {
         assertEquals(expectedState, uploadPictureSubmit.isEnabled());
         
+    }
+    
+    public void waitForUploadEditModalVisible() {
+        waitForElementVisibility(uploadEditModal);
     }
 
 }

